### DISCUSSION TITLE
prisma with postgis

### DISCUSSION THREAD

**Discussion by szmxx** [prisma/prisma#21823] [Q&A]
### Bug description

I have installed postgis and postgres and enabled the postgis extension for a certain database, but when I try to use `npx prisma migrate dev` it prompts me this ：
```
Database error code: 22023

Database error:
ERROR: extension "postgis" has no installation script nor update path for version "3.3"

DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E22023), message: "extension \"postgis\" has no installation script nor update path for version \"3.3\"", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("extension.c"), line: Some(1551), routine: Some("CreateExtensionInternal") }


 ELIFECYCLE  Command failed with exit code 1.
```

### How to reproduce

1、schema.prisma
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["postgresqlExtensions"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")

  extensions = [postgis(version: "3.3")]
}
```
2、npx prisma migrate dev 

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["postgresqlExtensions"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")

  extensions = [postgis(version: "3.3")]
}
```

```ts
// Add your code using Prisma Client
import { PrismaClient } from "@prisma/client";
let prisma: PrismaClient;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  let globalWithPrisma = global as typeof globalThis & {
    prisma: PrismaClient;
  };
  if (!globalWithPrisma.prisma) {
    globalWithPrisma.prisma = new PrismaClient();
  }
  prisma = globalWithPrisma.prisma;
}

export default prisma;

```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 5.5.2
@prisma/client          : 5.5.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/.pnpm/@prisma+engines@5.5.2/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/.pnpm/@prisma+engines@5.5.2/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.5.1-1.aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Default Engines Hash    : aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Studio                  : 0.494.0
Preview Features        : postgresqlExtensions
(base) 
```

**Comment by ludralph** [comment]
Hi @szmxx 👋 

Thank you for raising this question.

What version of PostGIS and PostgreSQL are you using? The error indicates that the version of the PostGIS extension installed in your PostgreSQL database does not have an installation script or update path for the version 3.3. This typically occurs when the PostGIS extension you have installed is not compatible with the version of PostgreSQL you are using.

**Reply by szmxx** [reply]
<img width="410" alt="image" src="https://github.com/prisma/prisma/assets/24505926/55f2a0a9-b618-4254-a05e-9ac1e1f9a6fa">


<img width="304" alt="image" src="https://github.com/prisma/prisma/assets/24505926/2a70a104-2d92-4fe7-89bf-e0c6c7f34e96">

postgres v15.3 and postgis v3.3

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by wintkhantlin** [comment]
Did you able to solve that I got that one too
